### ISSUE TITLE
bug report on version 0.10.2

### ISSUE THREAD

**Issue by andybrier** [influxdata/influxdb#5934]
version: V0.10.2 (STABLE)
operation system:  centos 6.5,   3.14.48-33.39.amzn1.x86_64 

```
unexpected fault address 0x7f8846c4c627
fatal error: fault
[signal 0xb code=0x1 addr=0x7f8846c4c627 pc=0x82eda1]

goroutine 2658085 [running]:
runtime.gothrow(0xbddb60, 0x5)
        /root/.gvm/gos/go1.4.3/src/runtime/panic.go:503 +0x8e fp=0xc217fa0d00 sp=0xc217fa0ce8
runtime.sigpanic()
        /root/.gvm/gos/go1.4.3/src/runtime/sigpanic_unix.go:29 +0x265 fp=0xc217fa0d50 sp=0xc217fa0d00
github.com/influxdb/influxdb/tsdb/engine/tsm1.btou16(0x7f8846c4c626, 0x2, 0x2fcab, 0xc2149a9f40)
        /tmp/influxdb/src/github.com/influxdb/influxdb/tsdb/engine/tsm1/writer.go:631 +0x31 fp=0xc217fa0d58 sp=0xc217fa0d50
github.com/influxdb/influxdb/tsdb/engine/tsm1.func·010(0x48e, 0xb0000c20814009c)
        /tmp/influxdb/src/github.com/influxdb/influxdb/tsdb/engine/tsm1/reader.go:422 +0xe0 fp=0xc217fa0de0 sp=0xc217fa0d58
sort.Search(0x91c, 0xc217fa0f00, 0xc210df4000)
        /root/.gvm/gos/go1.4.3/src/sort/search.go:66 +0x59 fp=0xc217fa0e10 sp=0xc217fa0de0
github.com/influxdb/influxdb/tsdb/engine/tsm1.(*indirectIndex).search(0xc2140dea00, 0xc210df4000, 0x9b, 0xa0, 0xa0)
        /tmp/influxdb/src/github.com/influxdb/influxdb/tsdb/engine/tsm1/reader.go:426 +0x69 fp=0xc217fa0f20 sp=0xc217fa0e10
github.com/influxdb/influxdb/tsdb/engine/tsm1.(*indirectIndex).Entries(0xc2140dea00, 0xc21c7495e0, 0x9b, 0x0, 0x0, 0x0)
        /tmp/influxdb/src/github.com/influxdb/influxdb/tsdb/engine/tsm1/reader.go:458 +0xf8 fp=0xc217fa1020 sp=0xc217fa0f20
github.com/influxdb/influxdb/tsdb/engine/tsm1.(*TSMReader).Entries(0xc21751cd20, 0xc21c7495e0, 0x9b, 0x0, 0x0, 0x0)
        /tmp/influxdb/src/github.com/influxdb/influxdb/tsdb/engine/tsm1/reader.go:290 +0x70 fp=0xc217fa1068 sp=0xc217fa1020
github.com/influxdb/influxdb/tsdb/engine/tsm1.(*FileStore).locations(0xc20a7eaf50, 0xc21c7495e0, 0x9b, 0xece7063fc, 0x8873d75, 0x1035bc0, 0x1, 0x0, 0x0, 0x0)
        /tmp/influxdb/src/github.com/influxdb/influxdb/tsdb/engine/tsm1/file_store.go:470 +0x3e9 fp=0xc217fa1290 sp=0xc217fa1068
github.com/influxdb/influxdb/tsdb/engine/tsm1.(*KeyCursor).init(0xc215622c40, 0xece7063fc, 0x8873d75, 0x1035bc0, 0x1)
        /tmp/influxdb/src/github.com/influxdb/influxdb/tsdb/engine/tsm1/file_store.go:558 +0xa5 fp=0xc217fa1300 sp=0xc217fa1290
github.com/influxdb/influxdb/tsdb/engine/tsm1.(*KeyCursor).SeekTo(0xc215622c40, 0xece7063fc, 0xe08873d75, 0x1035bc0, 0x1035b01, 0x0, 0x0, 0x0, 0x0, 0x0)
        /tmp/influxdb/src/github.com/influxdb/influxdb/tsdb/engine/tsm1/file_store.go:582 +0xa3 fp=0xc217fa1428 sp=0xc217fa1300
github.com/influxdb/influxdb/tsdb/engine/tsm1.(*devCursor).SeekTo(0xc21b15fa20, 0x1439c9772e1e1576, 0xffffffffffffffff, 0x0, 0x0)
        /tmp/influxdb/src/github.com/influxdb/influxdb/tsdb/engine/tsm1/engine.go:737 +0x173 fp=0xc217fa1530 sp=0xc217fa1428
github.com/influxdb/influxdb/tsdb.(*TagsCursor).SeekTo(0xc2160694a0, 0x1439c9772e1e1576, 0x0, 0x0, 0x0)
        /tmp/influxdb/src/github.com/influxdb/influxdb/tsdb/cursor.go:368 +0x61 fp=0xc217fa1570 sp=0xc217fa1530
github.com/influxdb/influxdb/tsdb.readMapInputs(0xc217bac000, 0x7b1f, 0x9800, 0xc213f4cf00, 0x1, 0x1, 0x1439c9772e1e1576, 0x1439c9772e1e1576, 0x1439c9779ccc6c00, 0x0, ...)
        /tmp/influxdb/src/github.com/influxdb/influxdb/tsdb/aggregate.go:835 +0x18e fp=0xc217fa1798 sp=0xc217fa1570
github.com/influxdb/influxdb/tsdb.(*AggregateMapper).NextChunk(0xc2080fa0f0, 0x0, 0x0, 0x0, 0x0)

```

**Comment by toddboom** [comment]
@andybrier Can you give us more info? How did this happen? Do you get the same error when you try to bring the service back up?

It looks like this might have been related to a query - if so, could you provide the query that triggered this?

Additionally, would you be able to share your data with us for further diagnosis?

**Comment by andybrier** [comment]
@toddboom  I restart, and get the same error now.  The detail file is here:
[influxd.log.txt](https://github.com/influxdata/influxdb/files/175779/influxd.log.txt)

**Comment by joelegasse** [comment]
@andybrier Can you provide us some more information on what kind of data you're writing to the server and how you are submitting it? How many series/tags/fields are you using, and what types of values are you storing the in the fields?

To reiterate @toddboom's question: would you be able to share with us a snapshot of your database?

**Comment by andybrier** [comment]
@joelegasse  I would like to share the snapshot of my database. but how can i get the snapshot?

**Comment by andybrier** [comment]
It happened about 2 or 3 days after i restarted the influxdb.

**Comment by joelegasse** [comment]
@andybrier You can just create an archive (tar or zip) of the directories in the database directory (`data` is the most important, but the others can be included, too)

**Comment by jwilder** [comment]
This is likely fixed in 0.11 via #6082.  The file causing the panic will need to be removed from your data directory since #6082 only fixes writing partial TSM files..
